lib/build.py

```diff
@@ -519,11 +519,13 @@ def execute_(self):
       # Expand source wildcards.
       srcs = list()
       for src in self.srcs:
-         for i in glob.glob("%s/%s" % (context, src)):  # glob can't take Path
+         matches = glob.glob("%s/%s" % (context, src))  # glob can't take Path
+         if (len(matches) == 0):
+            ch.FATAL("can't copy: not found: %s" % src)
+         for i in matches:
             srcs.append(i)
             ch.VERBOSE("source: %s" % i)
-      if (len(srcs) == 0):
-         ch.FATAL("can't COPY: no sources found")
+      assert (len(srcs) > 0)
```

Comment:
Do we think the extra line of code saved is worth the hairy traceback output from an assertion error? FWIW, I prefer the FATAL message to a messy traceback.

Reply:
In this case it really is a "can't happen" situation — it should have been caught by the test on line 523, but I felt nervous about it.

I do see now that if `self.srcs` has length zero then `srcs` will as well. Maybe that is why I felt nervous. Thanks for pointing it out.